samples/hello_fpga.c

```diff
@@ -230,6 +231,33 @@ fpga_result get_bus(fpga_token tok, uint8_t *bus)
 	return res1 != FPGA_OK ? res1 : res2;
 }
 
+/* Is the FPGA simulated with ASE? */
+bool probe_for_ase()
+{
+	fpga_result r = FPGA_OK;
+	uint16_t device_id = 0;
+	fpga_properties filter = NULL;
+	uint32_t num_matches = 1;
+	fpga_token fme_token;
+
+	/* Connect to the FPGA management engine */
+	fpgaGetProperties(NULL, &filter);
+	fpgaPropertiesSetObjectType(filter, FPGA_DEVICE);
+
+	/* Connecting to one is sufficient to find ASE */
+	fpgaEnumerate(&filter, 1, &fme_token, 1, &num_matches);
+	if (0 != num_matches) {
+		/* Retrieve the device ID of the FME */
+		fpgaGetProperties(fme_token, &filter);
```

Comment:
This call will allocate a new properties object, leaking the one allocated on 244.

Reply:
What's the right thing to do here? Call fpgaDestroyProperties() before 251 or some other call that adds the fme_token to the filter? If the latter, please give me a code snippet.

Reply:
To fix, destroy the original properties object before the 2nd call to fpgaGetProperties.

Reply:
Fixed now, I think. Is the updated code what you meant?

Reply:
Yes, that will do it.